### PR TITLE
Clean up diagnostic and solution crawler options - take 2

### PR DESCRIPTION
**Pull request by tmat**
Same as https://github.com/dotnet/roslyn/pull/59083 but last commit keeps the option that disables solution crawler for testing purposes.

The option is only considered during startup and changing the value later does not have any effect.

RPS validation:
- [x] https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/382767

### CODE REVIEW

File: src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs [hunk 1]
```diff
@@ -57,26 +57,23 @@ internal sealed partial class WorkCoordinator
...
                     activeFileBackOffTimeSpan, allFilesWorkerBackOffTimeSpan, entireProjectWorkerBackOffTimeSpan, _shutdownToken);
 
-                var semanticBackOffTimeSpan = InternalSolutionCrawlerOptions.SemanticChangeBackOffTimeSpan;
-                var projectBackOffTimeSpan = InternalSolutionCrawlerOptions.ProjectPropagationBackOffTimeSpan;
+                var semanticBackOffTimeSpan = SolutionCrawlerTimeSpan.SemanticChangeBackOff;
+                var projectBackOffTimeSpan = SolutionCrawlerTimeSpan.ProjectPropagationBackOff;
 
                 _semanticChangeProcessor = new SemanticChangeProcessor(listener, _registration, _documentAndProjectWorkerProcessor, semanticBackOffTimeSpan, projectBackOffTimeSpan, _shutdownToken);
 
                 // if option is on
```
Comment (JoeRobich) [inline src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs:73]: Remove or update stale comment.
```suggestion
```

File: src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerService.cs [hunk 2]
```diff
@@ -7,11 +7,17 @@
 using System.Composition;
 using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
 
 namespace Microsoft.CodeAnalysis.SolutionCrawler
 {
     internal partial class SolutionCrawlerRegistrationService : ISolutionCrawlerRegistrationService
     {
+        // TODO:
```
Comment (JoeRobich) [inline src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerService.cs:16]: What is there todo with the EnableSolutionCrawler option?

Comment (tmat) [reply src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerService.cs:16]: Not needed anymore. Removed.

**Comment by genlu** [comment]
I'm a little concerned about the refset regression, which might not be caused by outdated ibc data alone. looking at the refset stack, the diff is caused by solution crawler specific to project add:

image

Could you please make sure this is expected?

**Comment by tmat** [comment]
@genlu Should be addressed by the latest commit. Re-running validation.

**Review by JoeRobich — APPROVED** [review]

**Comment by RikkiGibson** [comment]
I couldn't find a validation PR for this which passed RPS. https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequests?_a=active&createdBy=9f64bc2f-479b-429f-a665-fec80e130b1f&assignedTo=82ac80ec-e1bc-496b-9112-c8b2e50f1324

The latest had some internal error that blocked meaningful results. https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/383173

I did find, though, that the first insertion that included these changes regresses RPS. https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/383755

Can we please refrain from merging things that were known to regress RPS in the past, until we have verified that they are no longer regressing RPS. @JoeRobich @tmat.

**Comment by tmat** [comment]
@genlu Confirmed that the regression in the linked validation PR: https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/382767 is not related to this change.

**Comment by RikkiGibson** [comment]
Here's the last known good: https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/383717

First known bad is this insertion already linked:
> The latest had some internal error that blocked meaningful results. https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/383173

The diff between the last known good and first known bad is this: https://github.com/dotnet/roslyn/compare/4280634ed99d64dd3fb01bc64d73145b1cb4535d...be10278d2ff469dfdceec2051af6a7caae5ea2f3?w=1

The failure persists through the subsequent insertions.
https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/383802
https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/383811

Do we think that Roslyn is perhaps not causing the regression at all here?

**Comment by tmat** [comment]
> Do we think that Roslyn is perhaps not causing the regression at all here?

That's possible. Looking at the traces again.

**Pull request merged**